### PR TITLE
RD-2592 - Update documentation verification script to verify both - frontend and backend README.md file

### PR DESCRIPTION
**Pull request by qooban**
Small update to verify-docs.sh script to verify both README.md files - for backend and frontend code.

Example of `npm run verify:docs` call (truncated some parts for better readability):

```
> cloudify-ui-common@4.0.0 verify:docs /home/qooban/WebstormProjects/cloudify-ui-common
> bash ci/verify-docs.sh

diff --git a/backend/README.md b/backend/README.md
index c888fe3..47ab509 100644
--- a/backend/README.md
+++ b/backend/README.md
@@ -2,45 +2,45 @@
 
 ### Table of Contents
 
--   [Database][1]
-    -   [DbInitializer][2]
-        -   [Parameters][3]
-    -   [init][4]

(...)

 #### Parameters
 
--   `loggerFactory` **[Object][15]** initialized logger factory, see [initLogging][7]
--   `dbModule` **[Object][15]** DB module, an object containing `init` function and `db` object, see [DbInitializer][2]
+*   `loggerFactory` **[Object][15]** initialized logger factory, see [initLogging][7]
+*   `dbModule` **[Object][15]** DB module, an object containing `init` function and `db` object, see [DbInitializer][2]
 
 [1]: #database
 
ERROR: JS backend API documentation not up-to-date with the source code. Follow these steps:
 - verify if 'backend/documentation.yml' configuration file is up-to-date with 'backend' folder content,
 - run 'npm run docs:backend',
 - check and commit updated README.md file.
npm ERR! code ELIFECYCLE

(...)

```

### CODE REVIEW

File: ci/verify-docs.sh [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+        echo " - check and commit updated README.md file."
+    fi
+
+    return $EXIT_CODE
+}
+
+checkReadmeFile "frontend" "src" "documentation.yml"; FRONTEND_CHECK_EXIT_CODE=$?
+checkReadmeFile "backend" "backend" "backend/documentation.yml"; BACKEND_CHECK_EXIT_CODE=$?
+
+! (( $FRONTEND_CHECK_EXIT_CODE || $BACKEND_CHECK_EXIT_CODE ))
```
Comment (Gelio) [inline ci/verify-docs.sh:24]: Not sure if we're not overcomplicating things :smile: We could just add `backend/README.md` to the `git diff` command

File: ci/verify-docs.sh [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
-  echo " - run 'npm run docs',"
-  echo " - check output 'src/README.md' file,"
-  echo " - commit updated 'src/README.md' file."
-  exit 1
-fi;
+
+checkReadmeFile() {
+    NPM_SCRIPT_SUFFIX=$1
+    README_DIRECTORY=$2
+    DOCS_CONFIG_FILE=$3
```
Comment (Gelio) [inline ci/verify-docs.sh:6]: Looks like magic :smile: I believe there are function calls in bash. Maybe using regular function calls would be less magical and easier to understand?

**Review by Gelio — APPROVED** [review]
I did not run the script myself and I do not have enough bash expertise to scan it that it works, but looks reasonable :+1:

File: ci/verify-docs.sh (hunk 1 above)
Comment (qooban) [reply ci/verify-docs.sh:24]: Yeah, I planned to do so originally, but after all to provide better (and more precise output about how to handle it) I decided to split it 🙂

File: ci/verify-docs.sh (hunk 2 above)
Comment (qooban) [reply ci/verify-docs.sh:6]: I'm also not expert in bash, I did it this way just to name expected function arguments 🙂

Comment (Gelio) [reply ci/verify-docs.sh:6]: Ok, I just found out that bash does not have named arguments :smile: 
https://linuxize.com/post/bash-functions/

Sounds reasonable, then!

Comment (qooban) [reply ci/verify-docs.sh:6]: Yes, unfortunately.

**Pull request merged**